components/lib/jexl/src/main/java/mozilla/components/lib/jexl/parser/Parser.kt

```diff
@@ -0,0 +1,250 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.lib.jexl.parser
+
+import mozilla.components.lib.jexl.ast.AstNode
+import mozilla.components.lib.jexl.ast.AstType
+import mozilla.components.lib.jexl.grammar.Grammar
+import mozilla.components.lib.jexl.lexer.Token
+
+/**
+ * JEXL parser.
+ *
+ * Takes a list of tokens from the lexer and transforms them into an abstract syntax tree (AST).
+ */
+internal class Parser(
```

Comment:
Lexer|Parser|Evaluator are all `internal`. Should they be? What is the main API entry point for other components?

Reply:
The public API is the `Jexl` class which allows consumers to evaluate expressions, add custom transforms and inject a custom/extended Grammar (not implemented yet).